packages/frontend/src/features/sqlRunner/transformers/useSqlChart.ts

```diff
@@ -0,0 +1,46 @@
+import {
+    BarChartDataTransformer,
+    isBarChartSQLConfig,
+    isPieChartSQLConfig,
+    PieChartDataTransformer,
+    type ResultRow,
+    type SqlColumn,
+    type SqlRunnerChartConfig,
+} from '@lightdash/common';
+import { useMemo } from 'react';
+import { useAsync } from 'react-use';
+import { SqlRunnerResultsTransformerFE } from './SqlRunnerResultsTransformerFE';
+
+export const useSqlChart = (
+    rows: ResultRow[],
+    columns: SqlColumn[],
+    config: SqlRunnerChartConfig,
+) => {
+    const transformer = useMemo(
+        () =>
+            new SqlRunnerResultsTransformerFE({
+                rows,
+                columns,
+            }),
+        [rows, columns],
+    );
+
+    const visTransformer = useMemo(() => {
+        if (isBarChartSQLConfig(config)) {
+            return new BarChartDataTransformer({
+                transformer,
+            });
+        } else if (isPieChartSQLConfig(config)) {
+            return new PieChartDataTransformer({
+                transformer,
+            });
+        } else {
+            throw new Error('Unknown chart type');
+        }
+    }, [config, transformer]);
+
+    return useAsync(
+        async () => visTransformer.getEchartsSpec(config),
+        [config, visTransformer],
+    );
```

Comment:
These 2 hooks can be combined as they have the same dependencies.

Reply:
You mean combine BarChartDataTransformer and PieChartDataTransformer? Can do in a follow up.